File: Illarion.Net/IDecoder.vb
Imports System.IO
Imports Illarion.Core.Types

Public Interface IDecoder
  Sub Shutdown()
  Function ReadString(stream As Stream) As String
  Function ReadByte(stream As Stream) As Byte
  Function ReadShort(stream As Stream) As Short
  Function ReadCharacterId(stream As Stream) As CharacterId
  Function ReadLocation(stream As Stream) As Location
End Interface

File: Network/Core/Interfaces/IDecoder.cs
using System.IO;
using Illarion.Common.Character;

namespace Illarion.Network.Core.Interfaces
{
    public interface IDecoder
    {
        void Shutdown();
        string ReadString(Stream stream);
        byte ReadByte(Stream stream);
        short ReadInt16(Stream stream);
        CharacterId ReadCharacterId(Stream stream);
        Coordinate ReadLocation(Stream stream);
    }
}